src/hotspot/os/linux/cgroupUtil_linux.cpp

```diff
@@ -46,3 +47,19 @@ int CgroupUtil::processor_count(CgroupCpuController* cpu_ctrl, int host_cpus) {
   log_trace(os, container)("OSContainer::active_processor_count: %d", result);
   return result;
 }
+
+CgroupMemoryController* CgroupUtil::adjust_controller(CgroupMemoryController* mem) {
+  if (mem->needs_hierarchy_adjustment()) {
+    julong phys_mem = os::Linux::physical_memory();
+    return mem->adjust_controller(phys_mem);
+  }
+  return mem;
+}
+
+CgroupCpuController* CgroupUtil::adjust_controller(CgroupCpuController* cpu) {
+  if (cpu->needs_hierarchy_adjustment()) {
+    int cpu_total = os::Linux::active_processor_count();
+    return cpu->adjust_controller(cpu_total);
+  }
+  return cpu;
```

Comment:
I guess an alternative - and maybe more readable solution - would be to inline `cpu->adjust_controller()` and `mem->adjust_controller()` code here. We have cg version agnostic api to query the limits. We'd just need accessors for `cgroup_path()` and a setter, `set_cgroup_path()` in `CgroupCpuController/CgroupMemoryController` impls.

Reply:
Currently (before my refactorization) the same code is duplicated 4 times. Do I understand the plan correctly it should be still duplicated but just 2 times? As without the duplication I do not see how to inline it.

Reply:
I have committed something, not sure if it is what was intended.